src/iuse.cpp

```diff
@@ -871,8 +887,17 @@ int iuse::poison(player *p, item *it, bool, const tripoint& )
 {
     if ((p->has_trait("EATDEAD"))) {
         return it->type->charges_to_use();
+    }
+
+    // NPCs have a magical sense of what is inedible
+    // Players can abuse the crafting menu instead...
+    if( !it->has_flag( "HIDDEN_POISON" ) &&
+        ( p->is_npc() ||
+          !p->query_yn( _("Are you sure you want to eat this? It looks poisonous...") ) ) ) {
```

Comment:
It specifically does not look poisonous, and the existence of a crafting menu exploit to test for it doesn't mean we should just nerf the feature into pointlessness.